### PR TITLE
Fix ordering functions for IFP

### PR DESCRIPTION
**Pull request by SwayStar123**
## Type of change

- Bug fix

## Changes

The IFP gt and lt functions were incorrectly implemented, this pr fixes the implementation and adds tests to ensure functionality

## Notes

- Bug was discovered here https://bugs.immunefi.com/dashboard/submission/32276

## Checklist

- [ ] I have linked to any relevant issues.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
   - [ ] If my change requires substantial documentation changes, I have [requested support from the DevRel team](https://github.com/FuelLabs/devrel-requests/issues/new/choose)
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.
- [x] I have updated the changelog to reflect the changes on this PR.

### CODE REVIEW

File: libs/src/fixed_point/ifp128.sw [hunk 1]
```diff
@@ -568,3 +568,66 @@ impl Power for IFP128 {
         }
     }
 }
+
+#[test]
+fn test_ord() {
```
Comment (bitzoic) [inline libs/src/fixed_point/ifp128.sw:573]: This needs to be moved to https://github.com/FuelLabs/sway-libs/blob/master/tests/src/fixed_point/ifp128_test/src/main.sw

Same applies to other tests

Comment (SwayStar123) [reply libs/src/fixed_point/ifp128.sw:573]: Actually i was gonna say those should be moved out of there. The structure makes no sense, its a sway script thats called by a rust sdk (and thats the only thing the sdk test does). Why are inline tests in the sdk tests section?

The only CI for sway inline tests is within the libs itself so i added the tests here. Should we make a seperate inline tests section in the tests folder?

**Review by bitzoic — APPROVED** [review]

**Review by K1-R1 — APPROVED** [review]

**Comment by bitzoic** [comment]
Dependent on https://github.com/FuelLabs/docs-hub/pull/340

**Comment by K1-R1** [comment]
Closing to re-open in order to try triggering stuck checks

**Pull request merged**